vendor/github.com/newrelic/infra-integrations-sdk/jmx/jmx.go

```diff
@@ -255,35 +324,49 @@ func Query(objectPattern string, timeout int) (map[string]interface{}, error) {
 
 // receiveResult checks for channels to receive result from nrjmx command.
 func receiveResult(lineCh chan []byte, queryErrors chan error, cancelFn context.CancelFunc, objectPattern string, timeout time.Duration) (result map[string]interface{}, err error) {
-	select {
-	case line := <-lineCh:
-		if line == nil {
+	gotResult := false
+	var warn string
+	for {
```

Comment:
If there are no errors or warnings, will we ever exit this for loop? It looks like if we receive a line, and it successfully sets `result`, this function won't exit until the timeout hits, which seems rather inefficient

Reply:
This exists on first error not being a *query one* as there might be several of these for a given bean pattern.

We added support for multiple lines on `nrjmx` so we cannot exit on first successful line. If you consider there's no reason to read a multiple line response from a query to `nrjmx` we could remove multiple line support again.

Once read all lines an error will be returned via `cmdErrC` causing loop to exit.